gpkit/interactive/widgets.py

```diff
@@ -77,40 +83,43 @@ def resolve(**subs):
                 sol = model.solution
             else:
                 sol = model.result
-            fn_of_sol(sol)
+            for fn in fns_of_sol:
+                fn(sol)
         except RuntimeWarning:
-            raise
-            # TODO: implement some nicer feasibility warning, like the below
-            # out = "THE PROBLEM IS INFEASIBLE"
-            # try:
-            #     const_feas = model.feasibility(["constants"])
-            #     out += "\n    but would become with this substitution:\n"
-            #     out += str(const_feas)
-            # except:
-            #     pass
-            # finally:
-            #     print(out)
+            print "Model did not solve!"
+            model.debug()
 
     resolve()
 
-    return widgets.interactive(resolve, **ranges)
+    return widgets.interactive(resolve, **ranges_out)
 
 
-def modelcontrolpanel(model, *args, **kwargs):
+# pylint: disable=too-many-locals
+def modelcontrolpanel(model, ranges=None, fns_of_sol=None, **solvekwargs):
     """Easy model control in IPython / Jupyter
 
     Like interact(), but with the ability to control sliders and their ranges
     live. args and kwargs are passed on to interact()
     """
 
-    sliders = model.interact(*args, **kwargs)
+    if fns_of_sol is None:
+        def __defaultfn(solution):
+            "Display function to run when a slider is moved."
+            tables = ["cost", "freevariables", "sensitivities"]
+            print solution.table(tables=tables)
+
+        __defaultfntable = __defaultfn
```

Comment:
Nope, because it needs to get replaced by the plots, which is done by function name.